python/strings_udf/cpp/src/strings/udf/shim.cu

```diff
@@ -270,3 +271,57 @@ extern "C" __device__ int rstrip(int* nb_retval,
 
   return 0;
 }
+
+extern "C" __device__ int upper(int* nb_retval,
+                                void* udf_str,
+                                void* const* st,
+                                std::int64_t flags_table,
+                                std::int64_t cases_table,
+                                std::int64_t special_table)
```

Comment:
Just curious why these are not `void*` as well?

Reply:
Let me make sure all is right here. Each of the functions that return a pointer to an input mapping table returns its own type of pointer. For the flags table its `uint8_t*`, for the cases table its `uint16_t`. For the special case mapping table it's a `special_case_mapping*`. Would the correct thing to do in this case be to receive each of these as a `uintptr_t` and then carry them through the python into the lowering as a `np.uintp`? Then these shim functions could accept a `uintptr_t` here.

Reply:
I think it would be more correct these to be pointer types than `int64_t` types.

Reply:
If you can interact with ctypes I think you can carry around a `ctypes.cvoidp`.